### PR TITLE
Requests module

### PR DESCRIPTION
**Pull request by mtarnawa**
In general the logical flow implies separate functions (not done here) for different operations.
Major things that were done are:

- requests module used - this allows acting upon HTTP responses other than 200/204 (in general successful
- eml files are downloaded in chunks (imagine something with a lot of attachments...)
- some small refactoring

BTW instead of specifying bound, you could find the full e-mail count for a mailbox with a separate request and then loop within the script. I see that the first GET requests returns in the payload:

` 'total_mailbox_mail_count': 16`

### CODE REVIEW

File: uidextract.py [hunk 1]
```diff
@@ -3,48 +3,61 @@
...
+    print("Getting list of emails")
+    # Create a session object from requests library
+    s = requests.Session()
+    retries = Retry(total=10, backoff_factor=1,
+                    status_forcelist=[500, 502, 504])
+    s.mount('https://', HTTPAdapter(max_retries=retries))
+    s.headers.update({'Cookie': 'csrftoken={0};'
+                      'sessionid={1}'.format(csrftoken, sessionid)})
+    mdatareq = 'https://app.openmailbox.org/requests/webmail?range={0}-{1}&sort=date&order=0&selected=&action=maillist&mailbox={2}'.format(lowerbound, upperbound, mailbox)
+    print(mdatareq)
```
Comment (mtarnawa) [inline uidextract.py]: Remove this though 🦆

**Comment by donat-b** [comment]
Just tested this, all messages downloaded without a problem.

**Comment by batterystaples** [comment]
Thanks! Merged.

**Comment by mtarnawa** [comment]
Thx. Maybe also update the readme so that people are aware to do `pip3 install -r requirements.txt`

**Comment by batterystaples** [comment]
Done! https://github.com/appleorange1/openmailbox_downloader/commit/f6ad0296de8d6d43936d4bd5cf4f183cded766ac

**Pull request merged**